feature-libs/customer-ticketing/components/customer-ticketing-create/customer-ticketing-create-dialog/customer-ticketing-create-dialog.component.ts

```diff
@@ -0,0 +1,106 @@
+import { Component, ElementRef, HostListener, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MAX_INPUT_CHARACTERS_FOR_MESSAGE } from '@spartacus/customer-ticketing/core';
+import { CustomerTicketingConfig } from '@spartacus/customer-ticketing/root';
+import {
+  FilesFormValidators,
+  FocusConfig,
+  FormUtils,
+  ICON_TYPE,
+  LaunchDialogService,
+} from '@spartacus/storefront';
+
+@Component({
+  selector: 'cx-customer-ticketing-dialog',
+  templateUrl: './customer-ticketing-create-dialog.component.html',
+})
+export class CustomerTicketingCreateDialogComponent implements OnInit {
+  iconTypes = ICON_TYPE;
+  form: FormGroup;
+  inputCharactersLimitForCreateMessage: number = this.getInputCharactersLimit;
+
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: 'button',
+    focusOnEscape: true,
+  };
+
+  get messagesCharacterLeft(): number {
+    return (
+      this.inputCharactersLimitForCreateMessage -
+      (this.form.get('message')?.value?.length || 0)
+    );
+  }
+
+  get allowedTypes(): string[] | undefined {
+    return this.customerTicketingConfig.customerTicketing
+      ?.attachmentRestrictions?.allowedTypes;
+  }
+
+  get getInputCharactersLimit(): number {
+    return (
+      this.customerTicketingConfig.customerTicketing
+        ?.inputCharactersLimitForCreateMessage ||
+      MAX_INPUT_CHARACTERS_FOR_MESSAGE
+    );
+  }
+
+  get maxSize(): number | undefined {
+    return this.customerTicketingConfig.customerTicketing
+      ?.attachmentRestrictions?.maxSize;
+  }
+
+  get maxEntries(): number | undefined {
+    return this.customerTicketingConfig.customerTicketing
+      ?.attachmentRestrictions?.maxEntries;
+  }
+
+  @HostListener('click', ['$event'])
+  handleClick(event: UIEvent): void {
+    if ((event.target as any).tagName === this.el.nativeElement.tagName) {
+      this.close('Click outside of the window');
+    }
+  }
+
+  constructor(
+    protected launchDialogService: LaunchDialogService,
+    protected el: ElementRef,
+    protected customerTicketingConfig: CustomerTicketingConfig,
+    protected filesFormValidators: FilesFormValidators
+  ) {}
+
+  ngOnInit(): void {
+    this.buildForm();
+  }
+
+  close(reason: string): void {
+    this.launchDialogService.closeDialog(reason);
+  }
+
+  createTicketRequest(): void {
+    if (this.form.invalid) {
+      this.form.markAllAsTouched();
+      FormUtils.deepUpdateValueAndValidity(this.form);
+    }
+  }
+
+  protected buildForm() {
+    const form = new FormGroup({});
+    form.setControl(
+      'message',
+      new FormControl('', [
+        Validators.required,
+        Validators.maxLength(this.inputCharactersLimitForCreateMessage),
+      ])
+    );
+    form.setControl(
+      'file',
+      new FormControl('', [
+        this.filesFormValidators.maxSize(this.maxSize),
+        // this.filesFormValidators.maçççxEntries(this.maxEntries),
```

Comment:
If not needed, please remove

Reply:
Should be kept, but will be normalized into a 'base' dialog that everyone will derive from 

@Parthlakhani